reaktive/src/commonMain/kotlin/com/badoo/reaktive/utils/serializer/DefaultSerializer.kt

```diff
@@ -0,0 +1,30 @@
+package com.badoo.reaktive.utils.serializer
+
+internal inline fun <T> serializer(
+    crossinline onValue: (T) -> Boolean,
+): Serializer<T> =
+    object : DefaultSerializer<T>() {
+        override fun onValue(value: T): Boolean =
+            onValue.invoke(value)
+    }
+
+internal abstract class DefaultSerializer<T> : AbstractSerializer<T>() {
+    private var queue: ArrayDeque<T>? = null
+
+    override fun addLast(value: T) {
+        if (queue == null) {
+            queue = ArrayDeque()
+        }
+        queue!!.add(value)
```

Comment:
> queue!!

Use same approach as in the other place please
`val q = queue ?: ArrayDeque<T>().also { queue = it }`